### PR TITLE
Added undeploy skip flag

### LINKED ISSUE #1358: Improvement to add skipUndeploy flag for undeploy goal
### Description
While it is currently supported to skip apply through the `skipApply` flag, I would like to have the similar flags for `undeploy`.

### Info

* Eclipse JKube version : 1.5.1
* Maven version (`mvn -v`) : 
```
Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739)
Maven home: /usr/local/Cellar/maven/3.8.3/libexec
Java version: 11.0.13, vendor: SAP SE, runtime: /Library/Java/JavaVirtualMachines/sapmachine-11.jdk/Contents/Home
Default locale: en_FR, platform encoding: US-ASCII
OS name: "mac os x", version: "12.2.1", arch: "x86_64", family: "mac"
```
* Kubernetes / Red Hat OpenShift setup and version : N/A

* Use case : 
  I am using multi-module project and I would like to apply/undeploy selectively certains modules based on maven configuration. The only way I can workaround is to the use the widder `skip` flag but I this would also skip build and resource goals which is not I want.

### PR DESCRIPTION
**Pull request by Prathamkrishna**
## Description

Fix #1358 

- Added undeploy skip flag in UndeployMojo
- Add skipUndeploy flag to KubernetesExtension in gradle plugins

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/develop/UndeployMojo.java [hunk 1]
```diff
@@ -83,4 +86,8 @@ protected List<File> getManifestsToUndeploy() {
     return ret;
   }
 
+  @Override
+  protected boolean canExecute() {
+    return super.canExecute() || skipUndeploy;
```
Comment (rohanKanojia) [inline kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/develop/UndeployMojo.java:91]: We don't know when we'll address https://github.com/eclipse/jkube/issues/1510 ; let's stick to this 

```suggestion
    return  super.canExecute() && !skipUndeploy;
```

**Comment by rohanKanojia** [comment]
Please add a test for the scenario when `skipUndeploy=true` in [UndeployMojoTest.java](https://github.com/eclipse/jkube/blob/master/kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/develop/UndeployMojoTest.java)

File: kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractJKubeMojo.java [hunk 2]
```diff
@@ -104,7 +104,7 @@ public abstract class AbstractJKubeMojo extends AbstractMojo implements KitLogge
     public void execute() throws MojoExecutionException, MojoFailureException {
         try {
             init();
-            if (!canExecute()) {
+            if (canExecute()) {
```
Comment (adriannowak) [inline kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractJKubeMojo.java:107]: The method name is not appropriate, or the log below is invalid. Method says that we 'canExecute' something, but log says we have skipped it

Comment (rohanKanojia) [reply kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractJKubeMojo.java:107]: You're right. We've created https://github.com/eclipse/jkube/issues/1510 in order to refactor this method's name

**Review by rohanKanojia — APPROVED** [review]

File: kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/develop/UndeployMojoTest.java [hunk 3]
```diff
@@ -47,6 +49,10 @@ public class UndeployMojoTest {
   private MavenProject mavenProject;
   @Mocked
   private Settings mavenSettings;
+  @Mocked
+  private KitLogger mockedKitLogger;
```
Comment (manusa) [inline kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/develop/UndeployMojoTest.java:53]: Can we use SilentLogger instead?

**Review by manusa — CHANGES_REQUESTED** [review]
LGTM, except for the minor change request

**Review by manusa — APPROVED** [review]

**Pull request merged**